mailbox/api/src/main/java/org/apache/james/mailbox/FlagsBuilder.java

```diff
@@ -21,14 +21,49 @@
 
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
 import javax.mail.Flags;
 
+import org.apache.james.util.StreamUtils;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableMap;
+
 public class FlagsBuilder {
 
     public static FlagsBuilder builder() {
         return new FlagsBuilder();
     }
 
+    public static List<String> getRawFlags(Flags flags) {
```

Comment:
As we have already a dozen transformation from Flag to String in James, it probably makes sense to describe how these Flags get encoded to Strings and have a testsuite for it.

Reply:
And I may have be wrong, maybe it's a serialization concern. What do others think about that ?

Reply:
There is only one place I've found flags <-> String transformation is in `MessageFlags`. But it only works with system flags.

Reply:
https://github.com/linagora/james-project/pull/2039#discussion_r242082496

> I think that the serialization layer is not responsible for knowing the system flags list.
We should keep this kind of logic in the Java code.

And then

> What I mean is that it probably belongs to the flagsbuilder, you should avoid any logic that is not simple translation between java and scala object.

To answer you concerns regarding Flags => List[String] in James code:
 - JMAP Keywords is rather specific, as the `\\` is not kept
 - This seems directly linked to backup flags writing: we do have the opportunity to factorize this... See  FlagsExtraField.
 - And of course we can see `ImapResponseComposerImpl::flags` (note that here flags order matters)

I could not find other places.

I would be in favor to factorize FlagsExtraField & Scala serialization.

True, we need to have a test suite for it.

I could not re

Reply:
Can we do it in separated PR? because this PR is becoming quite long, I want to merge it.

Reply:
Yes, please revert to the first version (where all the serialization logic is in scala), we'll discuss what is the best solution after this PR.

Reply:
rebased, revert to scala serialization.